src/yunohost/app.py

```diff
@@ -1366,34 +1451,66 @@ def _get_setting(settings, name):
+        redirected_regex.update(app_settings.get('redirected_regex', {}))
+
+        # Legacy permission system using (un)protected_uris and _regex managed in app settings...
+        unprotected_urls += [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'unprotected_uris')]
+        protected_urls += [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'protected_uris')]
+        unprotected_regex += _get_setting(app_settings, 'unprotected_regex')
+        protected_regex += _get_setting(app_settings, 'protected_regex')
+
+        # New permission system
+        this_app_perms = {name: info for name, info in all_permissions.items() if name.startswith(app['id'] + ".")}
+        for perm_name, perm_info in this_app_perms.items():
+            # FIXME : gotta handle regex-urls here... meh
+            urls = [_sanitized_absolute_url(url) for url in perm_info["urls"]]
+            if "visitors" in perm_info["allowed"]:
+                unprotected_urls += urls
+
+                # Legacy stuff : we remove now unprotected-urls that might have been declared as protected earlier...
+                protected_urls = [u for u in protected_urls if u not in urls]
+            else:
+                # TODO : small optimization to implement : we don't need to explictly add all the app roots
 
-            for item in _get_setting(app_settings, 'skipped_uris'):
-                if item[-1:] == '/':
-                    item = item[:-1]
-                skipped_urls.append(app_settings['domain'] + app_settings['path'].rstrip('/') + item)
-            for item in _get_setting(app_settings, 'skipped_regex'):
-                skipped_regex.append(item)
-            for item in _get_setting(app_settings, 'unprotected_uris'):
-                if item[-1:] == '/':
-                    item = item[:-1]
-                unprotected_urls.append(app_settings['domain'] + app_settings['path'].rstrip('/') + item)
-            for item in _get_setting(app_settings, 'unprotected_regex'):
-                unprotected_regex.append(item)
-            for item in _get_setting(app_settings, 'protected_uris'):
-                if item[-1:] == '/':
-                    item = item[:-1]
-                protected_urls.append(app_settings['domain'] + app_settings['path'].rstrip('/') + item)
-            for item in _get_setting(app_settings, 'protected_regex'):
-                protected_regex.append(item)
-            if 'redirected_urls' in app_settings:
-                redirected_urls.update(app_settings['redirected_urls'])
-            if 'redirected_regex' in app_settings:
-                redirected_regex.update(app_settings['redirected_regex'])
+                protected_urls += urls
+
+                # Legacy stuff : we remove now unprotected-urls that might have been declared as protected earlier...
+                unprotected_urls = [u for u in unprotected_urls if u not in urls]
```

Comment:
Are you sure of that ? :-/ At line 1506 OK, but here I don't understand why...

Reply:
Hmmm, idk, don't remember exactly, but basically if add an url as protected, it shouldn't be "unprotected" by legacy stuff ... E.g. say that you install an app still using the legacy stuff as public : 
- it will have declared `unprotected_urls = /` in its setting
- now the user use the permission system to declare that it shouldn't be public anymore
- so now the new permission system will set `/` as protected
- but without the line we're discussing, `/` will also be in the unprotected urls ...

Reply:
Ok, but why we don't cleanup the settings and remove all old permission things ?

Reply:
Well reading your other answer I understand...

Reply:
> Hmmm, idk, don't remember exactly, but basically if add an url as protected, it shouldn't be "unprotected" by legacy stuff ... E.g. say that you install an app still using the legacy stuff as public :
>    - it will have declared `unprotected_urls = /` in its setting
>    - now the user use the permission system to declare that it shouldn't be public anymore
>    - so now the new permission system will set `/` as protected
>    - but without the line we're discussing, `/` will also be in the unprotected urls ...

For me there are 2 cases:
- The permission was migrated automatically. In this case we can use the permission to change the access to the app. And in this case we don't need this line.
- The permission wasn't migrated because it was too customized. in this case the user is not able to change the permission (because it was not migrated). So we don't have any permission for this custom case in LDAP. So this line in this case is also not needed.

Reply:
>The permission was migrated automatically. In this case we can use the permission to change the access to the app. And in this case we don't need this line.

It's fuzzy to me, but I think that we still need to keep the line, because as said in the previous comment, if the app install declares `unprotected_urls = /` (so the app is public) then the user changes the permission to private using the new permission system, then the line `unprotected_urls = /` is **still there** so we need to remove it from the "compiled" ssowat configuration if there's a conflict with the new permission system ...